cluster-autoscaler/cloudprovider/clusterapi/clusterapi_test_framework.go

```diff
@@ -0,0 +1,760 @@
+				"clusterName":           spec.clusterName,
+				"machineDeploymentName": spec.machineDeploymentName,
+			}
+
+			config.machineDeployment = &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"kind":       machineDeploymentKind,
+					"apiVersion": "cluster.x-k8s.io/v1beta2",
+					"metadata": map[string]interface{}{
+						"name":      spec.machineDeploymentName,
+						"namespace": spec.namespace,
+						"uid":       spec.machineDeploymentName,
+					},
+					"spec": map[string]interface{}{
+						"clusterName": spec.clusterName,
+						"replicas":    int64(spec.nodeCount),
+						"template": map[string]interface{}{
+							"spec": map[string]interface{}{
+								"infrastructureRef": map[string]interface{}{
+									"apiGroup": "infrastructure.cluster.x-k8s.io",
+									"kind":     machineTemplateKind,
+									"name":     "TestMachineTemplate",
+								},
+							},
+						},
+					},
+					"status": map[string]interface{}{},
+				},
+			}
+			config.machineDeployment.SetAnnotations(spec.annotations)
+			config.machineDeployment.SetLabels(machineDeploymentLabels)
+			if err := unstructured.SetNestedStringMap(config.machineDeployment.Object, machineDeploymentLabels, "spec", "selector", "matchLabels"); err != nil {
+				panic(err)
+			}
+
+			ownerRefs := []metav1.OwnerReference{
+				{
+					Name: config.machineDeployment.GetName(),
+					Kind: config.machineDeployment.GetKind(),
+					UID:  config.machineDeployment.GetUID(),
+				},
+			}
+			config.machineSet.SetOwnerReferences(ownerRefs)
+		}
+		config.machineSet.SetLabels(machineSetLabels)
+		if err := unstructured.SetNestedStringMap(config.machineSet.Object, machineSetLabels, "spec", "selector", "matchLabels"); err != nil {
+			panic(err)
+		}
+
+		machineOwner := metav1.OwnerReference{
+			Name: config.machineSet.GetName(),
+			Kind: config.machineSet.GetKind(),
+			UID:  config.machineSet.GetUID(),
+		}
+
+		if spec.capacity != nil {
+			klog.V(4).Infof("adding capacity to machine template")
+			config.machineTemplate = &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "infrastructure.cluster.x-k8s.io/v1beta1",
+					"kind":       machineTemplateKind,
+					"metadata": map[string]interface{}{
+						"name":      "TestMachineTemplate",
+						"namespace": spec.namespace,
+						"uid":       "TestMachineTemplate",
+					},
+				},
+			}
+			if err := unstructured.SetNestedStringMap(config.machineTemplate.Object, spec.capacity, "status", "capacity"); err != nil {
+				panic(err)
+			}
+		} else {
+			klog.V(4).Infof("not adding capacity")
+		}
+
+		for j := 0; j < spec.nodeCount; j++ {
+			config.nodes[j], config.machines[j] = makeLinkedNodeAndMachine(j, spec.namespace, spec.clusterName, machineOwner, machineSetLabels)
+		}
+
+		result = append(result, config)
+	}
+
+	return result
+}
+
+// TestSpec contains the clusterapi specific information for a single test specification.
+type TestSpec struct {
```

Comment:
This looks identical to the existing `testSpec` type definition, why do we want to make it exportable?